### PR TITLE
fix(widget-list):  when single field value is object widget

### LINKED ISSUE #2375: List of object is not editable.
Hi, when I create list of objects, fields are not editable. 
Not working with `field: Object`
```
      - label: "Sections"
        name: "sections"
        widget: "list"
        field:
          label: "Author"
          name: "author"
          widget: "object"
          fields:
            - { label: "Navigation text", name: "navtitle", widget: "text" }
            - label: "Type"
              name: "type_ref"
              widget: "select"
              options:
                - { label: "Title", value: "string" }
                - { label: "Text", value: "text" }
                - { label: "Checker", value: "boolean" }
            - {label: "Value", name: "value", widget: "string"}

```
If used `fields: - Object`. everything works as desired. I believe this is an bug how the field: is proccessed by cms. 
```
      - label: "Sections"
        name: "sections"
        widget: "list"
        fields:
         - label: "Author"
           name: "author"
           widget: "object"
           fields:
             - { label: "Navigation text", name: "navtitle", widget: "text" }
             - label: "Type"
               name: "type_ref"
               widget: "select"
               options:
                 - { label: "Title", value: "string" }
                 - { label: "Text", value: "text" }
                 - { label: "Checker", value: "boolean" }
                 - {label: "Value", name: "value", widget: "string"}

```

### PR DESCRIPTION
**Pull request by barthc**
**Summary**
Closes #2375

### CODE REVIEW

**Comment by barthc** [comment]
@erquhart  ping!!!

File: packages/netlify-cms-widget-list/src/ListControl.js [hunk 1]
```diff
@@ -200,10 +200,13 @@ export default class ListControl extends React.Component {
     return (fieldName, newValue, newMetadata) => {
       const { value, metadata, onChange, field } = this.props;
       const collectionName = field.get('name');
-      const newObjectValue =
-        this.getValueType() !== valueTypes.SINGLE
-          ? this.getObjectValue(index).set(fieldName, newValue)
-          : newValue;
+      const listFieldObjectWidget = field.getIn(['field', 'widget']) === 'object';
```
Comment (erquhart) [inline packages/netlify-cms-widget-list/src/ListControl.js]: I would expect `field.widget` to almost always be `"list"` - what am I missing?

Comment (barthc) [reply packages/netlify-cms-widget-list/src/ListControl.js]: The fix here is to save the value just like we do for `list`  multiple `fields`  i.e using widget name as key  `{widget_name: value}`.
So `this.getValueType() === valueTypes.SINGLE && listFieldObjectWidget` should match config setup as described in the issue.

**Comment by barthc** [comment]
@erquhart  still merging this PR?

**Comment by erquhart** [comment]
I think my issue with this was hardcoding the name of the object widget, but our Widget API doesn't really provide a better way - this is probably the best move for now. Sorry for the delay on this @barthc!

**Pull request merged**